src/vm/tests/simple_apply_eval.rs

```diff
@@ -911,6 +911,18 @@ fn test_lets() {
         .for_each(|(program, expectation)| assert_eq!(expectation.clone(), execute(program)));
 }
 
+#[test]
+fn test_2053_stacked_user_funcs() {
+    let test = "
+(define-read-only (identity (n int)) n)
+(begin (identity (identity 1)))
+";
+
+    let expectation = Value::Int(1);
+
+    assert_eq!(expectation, execute(test));
+}
```

Comment:
With the changes to how we measure stack depth, the VM would rightfully prevent us from doing something like `(begin (identity (identity (identity (identity ( ... (identity 1))))))) ... )`, right?

Reply:
Yep -- it prevented this before this PR, but the way this was enforced was what was causing this problem. The tests:

```
    vm::tests::contracts::test_arg_stack_depth
    vm::tests::contracts::test_cc_stack_depth
```

demonstrate that prevention.

Reply:
In order to preserve that behavior, we needed to add the `apply_depth` value.